### PR TITLE
Contact test

### PR DESCRIPTION
**Pull request by Scieon**
3/4 of the tests fail because we cannot test modal popup.
I still included the failed test because they give us 96% coverage.

### CODE REVIEW

File: src/pages/contact/contact.html [hunk 1]
```diff
@@ -20,7 +20,7 @@
   <ion-list>
     <ion-item *ngFor="let contact of contacts" (click)="showContactModal(contact)">
       <h2>{{contact.name}}</h2>
-      <p>{{contact.phone_number | phone}}</p>
+      <p>{{contact.phone_number}}</p>
```
Comment (Scieon) [inline src/pages/contact/contact.html:23]: Removed pipe cause it affected the unit test, it also did not make a difference to the UI. Let me know if it is needed.

Comment (JTalarico) [reply src/pages/contact/contact.html:23]: the phone numbers are piped in the back end now and so this was redundant. Nice find.

**Review by ghiobi — APPROVED** [review]
Thank you anhkhoi
> Program testing can be used to show the presence of bugs, but never to show their absence! - Edsger W. Dijkstra

File: src/pages/contact/contact.spec.ts [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+      declarations: [MyApp, ContactPage],
+      imports: [
+        IonicModule.forRoot(ContactPage),
+      ],
+      providers: [
+        {
+          deps: [MockBackend, BaseRequestOptions],
+          provide: Http, useFactory: (backend, options) => {
+          return new Http(backend, options);
+        },
```
Comment (JTalarico) [inline src/pages/contact/contact.spec.ts:52]: return should have an extra indentation along with the bracket.

File: src/pages/contact/contact.spec.ts [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+  })));
+
+  it("Add contact modal should be created", async(inject([ContactsService], (service) => {
+    expect(component.showAddContactModal()).toBeDefined();
+  })));
+
+  it("Contacts should be filtered", async(inject([ContactsService], (service) => {
+    expect(component.filterContacts(new MockEvent())).toBeDefined();
+  })));
+
```
Comment (JTalarico) [inline src/pages/contact/contact.spec.ts:97]: It seems like you check every aspect of contacts. Nice job, having robust testing is very important.

File: src/pages/contact/contact.spec.ts (hunk 2 above)
Comment (Scieon) [reply src/pages/contact/contact.spec.ts:52]: This is correct indentation following Ratliff style according to IDE

**Pull request merged**